sdk/core/azure-core/src/main/java/com/azure/core/util/ETag.java

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.util;
+
+import com.azure.core.util.logging.ClientLogger;
+
+/**
+ * This class represents an HTTP ETag. A ETag value could be strong or weak ETag.
+ * More information, check https://en.wikipedia.org/wiki/HTTP_ETag
+ */
+public final class ETag {
```

Comment:
Another potential option here would be instead of making this an instantiable class we make this a utility class with two methods, maybe three.

```
public static String createETag(String eTag);

public static String createWeakETag(String eTag);

public static boolean compareETags(String eTag1, String eTag2);
```

 An a constant representing the "ALL" ETag (`*`).

This would be easier to integrate into currently existing APIs as they wouldn't need a new overload which takes `ETag` but instead can reference this class as a utility to help construct ETags.

cc: @JonathanGiles

Reply:
I tend to agree, but I am not sure of the degree of value that the utility methods bring?

Reply:
After offline discuss with Alan. We think it is better to keep as it is.